### PR TITLE
[WIP] New event to retrieve the widget class

### PR DESCRIPTION
**Pull request by richardhj**
use case follows here

### CODE REVIEW

**Comment by richardhj** [comment]
Please reopen

**Comment by richardhj** [comment]
This PR will get RFC *after* I checked if this PR is suitable for MetaModels/contao-frontend-editing#8

**Comment by discordier** [comment]
@richardhj Is it really still WIP? If not, feel free to merge.
I mean, we have almost lost this PR during two merges already.

File: src/ContaoCommunityAlliance/DcGeneral/Contao/View/Contao2BackendView/Subscriber/WidgetBuilder.php [hunk 1]
```diff
@@ -102,29 +105,21 @@ public function getEnvironment()
...
     {
-        if (isset($this->widgetMapping[$property->getWidgetType()])) {
-            return $this->widgetMapping[$property->getWidgetType()];
-        }
-
-        if (!isset($GLOBALS['BE_FFL'][$property->getWidgetType()])) {
-            return null;
-        }
+        $dispatcher = $environment->getEventDispatcher();
+        $dispatcher->addListener(DcGeneralEvents::GET_WIDGET_CLASS, [$this, 'retrieveWidgetClass']);
```
Comment (discordier) [inline src/ContaoCommunityAlliance/DcGeneral/Contao/View/Contao2BackendView/Subscriber/WidgetBuilder.php]: I don't like this very much.

The event dispatcher must be immutable (at least for Contao 4) and therefore we should rather use this as fallback if the class is still `null` after dispatching.

**Pull request closed without merging**